### PR TITLE
Fix --wait issue for eksctl upgrade nodegroup command

### LINKED ISSUE #3889: --wait=false does not work when upgrading nodegroups
When using eksctl 0.54.0 to update a managed node group from 1.19 to 1.20, the --wait=false flag still causes the command to wait for CloudFormation.

When using the --timeout flag that will make it only wait for the time specified, but the exit code is 1, causing it to fail any scripts I use the command in.

This is the command I used:
`eksctl upgrade nodegroup --cluster cluster-name --name=nodegroup-name --wait=false --kubernetes-version=1.20`

### PR DESCRIPTION
**Pull request by vchetu**
Fixes #3889 

### Description

Currently --wait flag is ignored while performing nodegroup upgrade action. This PR checks for wait flag and accordingly performs the action.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by nikimanoledaki** [comment]
Thank you for opening a PR @vchetu, a member of the team will review shortly!

File: pkg/actions/nodegroup/upgrade.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ func (m *Manager) Upgrade(options managed.UpgradeOptions) error {
 		return managedService.UpgradeNodeGroup(options)
 	}
 
-	return m.upgradeAndWait(options)
+	return m.upgradeAndWait(options, wait)
```
Comment (cPu1) [inline pkg/actions/nodegroup/upgrade.go:37]: A cleaner way would be to split `upgradeAndWait` into two functions `upgrade` and `waitForUpgrade`, and call the latter one only if `wait` is set:

```suggestion
	if err := m.upgrade(options); err != nil {
		return err
	}
	if wait {
		return m.waitForUpgrade(options)
	}
	return nil
```

**Review by cPu1 — CHANGES_REQUESTED** [review]

File: pkg/actions/nodegroup/upgrade.go (hunk 1 above)
Comment (vchetu) [reply pkg/actions/nodegroup/upgrade.go]: Thanks @cPu1. Modified the code accordingly

File: pkg/actions/nodegroup/upgrade.go [hunk 2]
```diff
@@ -34,10 +34,19 @@ func (m *Manager) Upgrade(options managed.UpgradeOptions) error {
...
-	return m.upgradeAndWait(options)
+	if err := m.upgrade(options); err != nil {
+		return err
+	}
+
+	if wait {
+		return m.waitForUpgrade(options)
+	}
+
+	return nil
```
Comment (cPu1) [inline pkg/actions/nodegroup/upgrade.go:45]: We should also log that the request has been submitted if `wait` is not set:

```suggestion
	logger.Info("nodegroup upgrade request submitted successfully")
	return nil
```

**Comment by cPu1** [comment]
Thanks for the PR, @vchetu. It needs just one more change and it's good to merge.

File: pkg/actions/nodegroup/upgrade.go (hunk 2 above)
Comment (vchetu) [reply pkg/actions/nodegroup/upgrade.go:47]: @cPu1 Added the log info.

**Review by cPu1 — APPROVED** [review]

**Pull request merged**